packages/std/src/binary.rs

```diff
@@ -30,6 +31,49 @@ impl Binary {
     pub fn as_slice(&self) -> &[u8] {
         self.0.as_slice()
     }
+
+    /// Copies content into fixed-sized array.
+    /// The result type `A: ByteArray` is a workaround for
+    /// the missing [const-generics](https://rust-lang.github.io/rfcs/2000-const-generics.html).
+    /// `A` is a fixed-sized array like `[u8; 8]`.
+    ///
+    /// ByteArray is implemented for `[u8; 0]` to `[u8; 64]`, such that
+    /// we are limited by 64 bytes for now.
+    ///
+    /// # Examples
+    ///
+    /// Copy to array of explicit length
+    ///
+    /// ```
+    /// # use cosmwasm_std::Binary;
+    /// let binary = Binary::from(&[0xfb, 0x1f, 0x37]);
+    /// let array: [u8; 3] = binary.to_array().unwrap();
+    /// assert_eq!(array, [0xfb, 0x1f, 0x37]);
+    /// ```
+    ///
+    /// Copy to integer
+    ///
+    /// ```
+    /// # use cosmwasm_std::Binary;
+    /// let binary = Binary::from(&[0x8b, 0x67, 0x64, 0x84, 0xb5, 0xfb, 0x1f, 0x37]);
+    /// let num = u64::from_be_bytes(binary.to_array().unwrap());
```

Comment:
Yeah, I've had some issues trying to map the types from `be_bytes` from slices.

While you're at it, do you want to implement the info version with this magic `ByteArray` trait? (not sure if this is actually useful).

```rust
// idea
let bin: Binary = 1000u64.to_be_bytes().into();
// currently this works
let bin: Binary = 1000u64.to_be_bytes().to_vec().into();
```

Actually, probably no good case to add this code.

Reply:
This is already implemented here: https://github.com/CosmWasm/cosmwasm/blob/v0.11.1/packages/std/src/encoding.rs#L60-L89